enum.rb

```diff
@@ -0,0 +1,53 @@
+#:nodoc:
+module Enumerable
+  def my_each
+    length.times { |index| yield(self[index]) }
+    self
+  end
+
+  def my_each_with_index
+    length.times { |index| yield(self[index], index) }
+    self
+  end
+
+  def my_select
+    array = []
+    my_each { |index| array << index if yield(index) }
+    array
+  end
+
+  def my_all?
+    new_array = true
+    my_each { |index| break new_array = false unless yield(index) }
+    new_array
+  end
+
+  def my_any?
+    new_array = false
+    my_each { |index| break new_array = true if yield(index) }
+    new_array
+  end
+
+  def my_none?
+    new_array = true
+    my_each { |index| break new_array = false if yield(index) }
+    new_array
+  end
+
+  def my_count(parameter)
+    count = 0
+    my_each { |index| count += 1 if parameter == index }
+    count
+  end
+
+  def my_map(proc)
+    mapped = []
```

Comment:
### Should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
``` p [1,2,3,4].map { |li| li*4 } ``` should return same results as ```p [1,2,3,4].map { |li| li*4 }```